include/ops.h

```diff
@@ -162,5 +165,17 @@ namespace std
         return static_cast<std::size_t>(o);
       }
     };
+
+  // specialize the hash template for Op
+  template<>
+    struct hash<smt::Op>
+    {
+      size_t operator()(const smt::Op o) const
+      {
+        // The hash function for op computes the string hash
+        std::hash<std::string> str_hash;
+        return str_hash(o.to_string());
```

Comment:
One comment: if we ever decide this hash is too slow (which I doubt), then we could always just default to using the hash of the `PrimOp`. That actually seems like a reasonable thing to do to me, because there's only a few indexed operators, and I don't think it's so bad if those are all kept in the same table entry, e.g. you'd have:
`hash(Extract) -> [(_ extract 0 0), (_ extract 8 4), ...]`

Not saying we need to make that change, just mentioning it.

Reply:
Yes I agree that this can be optimized, but also doubt it will become a bottleneck (it is always O(1), except for maybe crazy uninterpreted sorts with arities > 1 built on top of each other).